sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -77,8 +82,54 @@ object CurrentOrigin {
 // A tag of a `TreeNode`, which defines name and type
 case class TreeNodeTag[T](name: String)
 
+// A wrapper of BitSet for pattern enums.
+trait TreePatternBits {
```

Comment:
Nit: how about putting TreePatternBits in a new file. There are already 1k lines code in `TreeNode.scala`

Reply:
Done.